### DISCUSSION TITLE
Support deprecated `CompletionItemTag` when importing objects marked as `@deprecated`

### DISCUSSION THREAD

**Discussion by Viicos** [microsoft/pylance-release#6179] [Enhancement]
I don't know if this is possible for performance reasons, but it would be nice to have imported objects marked as [`@deprecated`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#completionItemTag) using the [`CompletionItemTag.Deprecated`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#completionItemTag) LSP tag.

Additionally, such deprecated items could be suggested last, to have a better auto-completion experience: 

![image](https://github.com/user-attachments/assets/b1271253-ba04-417c-9b6e-2429ac8b8420)

**Comment by disrupted** [comment]
created an issue for Pyright as well https://github.com/microsoft/pyright/issues/9915

**Comment by disrupted** [comment]
FYI this feature is now available in the latest basedpyright release https://github.com/DetachHead/basedpyright/releases/tag/v1.28.0
